transport/transport.go

```diff
@@ -306,3 +319,71 @@ func NewMultiTransport(backends []http.RoundTripper,
 		Backends:        backends,
 		HandleResponses: responsesHandler}
 }
+
+// SetTransportsConfig assign transport config to Matcher
+func (m *Matcher) SetTransportsConfig(clientConfig httphandlerConfig.Client) {
+	m.TransportsConfig = clientConfig.Transports
+}
+
+// SelectTransport returns transport name by method, path and queryParams
+func (m *Matcher) SelectTransport(method, path, queryParams string) (transportName string) {
+	_, transportName, ok := m.TransportsConfig.GetMatchedTransport(method, path, queryParams)
+	if !ok {
+		log.DefaultLogger.Fatalf("Transport not matched with args. method: %s, path: %s, queryParams: %s", method, path, queryParams)
+	}
+	return
+}
+
+// RoundTrip extends TransportRoundTripper struct wtih RoundTripper and transports container
+func (m *Matcher) RoundTrip(request *http.Request) (*http.Response, error) {
+	return m.SelectTransportByRequest(request).RoundTrip(request)
+}
+
+// SelectTransportByRequest for selecting RoundTripper by request object from transports container
+func (m *Matcher) SelectTransportByRequest(request *http.Request) (selectedRoundTripper http.RoundTripper) {
+	selectedTransportName := m.SelectTransport(request.Method, request.URL.Path, request.URL.RawQuery)
+	reqID := request.Context().Value(log.ContextreqIDKey)
+	log.DefaultLogger.Debugf("Request %s - selected transport name: %s (by method: %s, path: %s, queryParams: %s)",
+		reqID, selectedTransportName, request.Method, request.URL.Path, request.URL.RawQuery)
+
+	return m.RoundTrippers[selectedTransportName]
```

Comment:
I don't understand why SelectTransport does return transport name instead of transport instance. If would do the latter this mapping call could be omitted. Transport name is one of it's properties